app/src/pages/Editor/Cursor.tsx

```diff
@@ -15,14 +15,27 @@ export function Cursor(): JSX.Element {
   const theme = useTheme();
 
   const [ref, setRef] = useState<HTMLDivElement | null>(null);
+  const now = Date.now();
+  const [scrollBlockedLast, setScrollBlockedLast] = useState(now);
 
   // we do this to re-render the cursor when the parent container size changes
   const _parentSize = useElementSize(ref?.parentElement);
 
-  const { left, top } = useComputeCursorPosition() || {
-    left: -100,
-    top: -100,
-  };
+  const cursorPosition = useComputeCursorPosition();
+  if (cursorPosition == null) return <></>;
+
+  const scrollContainer = document.getElementById('scroll-container');
+  if (
+    scrollContainer &&
+    now - scrollBlockedLast > 1000 &&
+    (scrollContainer.scrollTop > cursorPosition.top - 20 ||
+      scrollContainer.scrollTop + scrollContainer.clientHeight - 20 < cursorPosition.top)
+  ) {
+    setTimeout(() => {
+      scrollContainer.scrollTo({ top: cursorPosition.top - 100, behavior: 'smooth' });
```

Comment:
Why -100?

Reply:
to not have the cursor be at the absolute top of the screen but rather have some padding above. 100 is just some arbitrary value that defines that padding

Reply:
this is now 30px which is ~one line